conjure-undertow-annotations/src/main/java/com/palantir/conjure/java/undertow/annotations/ParamDecoders.java

```diff
@@ -252,6 +253,36 @@
         return DelegatingCollectionParamDecoder.of(serde::deserializeUuidSet);
     }
 
+    public static <T> ParamDecoder<T> complexParamDecoder(PlainSerDe serde, Function<String, T> factory) {
+        return DelegatingParamDecoder.of(value -> serde.deserializeComplex(value, factory));
+    }
+
+    public static <T> ParamDecoder<Optional<T>> optionalComplexParamDecoder(
+            PlainSerDe serde, Function<String, T> factory) {
+        return DelegatingParamDecoder.of(value -> serde.deserializeOptionalComplex(
+                value == null ? ImmutableList.of() : ImmutableList.of(value), factory));
```

Comment:
This one doesn't have a non-collection input, so the implementation slightly diverges

Reply:
I'm not sure it's possible to hit the optional.empty case here